src/System.Text.Json/src/System/Text/Json/JsonConstants.cs

```diff
@@ -63,7 +63,7 @@ internal static class JsonConstants
 
         public const int MaxEscapedTokenSize = 1_000_000_000;   // Max size for already escaped value.
```

Comment:
Where does this value come from?

Reply:
We compute for the maximum required space to write a property name and value string, and then ask for that size'd buffer from the `IBufferWriter<byte>` (the call to `GetSpan(sizeHint)`).

What's the largest key/value pairs that can fit after making a single call to GetSpan(...)?
Say, you want to write: "really large property name": " really large value". You do this as follows:
```C#
// This needs to call IBW.GetSpan with the required size to write this data into it
jsonWriter.WriteString("really large property name", " really large value");
```

The maximum feasible is roughly: int.MaxValue - 56 (for array-backed buffer)
Then we have up to 1000 depth, so account for indentation (and quotes, spaces, colon, etc), so let's round the maximum feasible down to 2 billion.
We have 1 billion for the property name string, and 1 billion for the value string. That's where this limit comes from.

Reply:
Sure, but why not just ask the IBufferWriter for what's needed, and let it fail if it can't provide the space?

Reply:
Does our read implementation have some sanity limits that we're trying to keep our write implementation under?

Reply:
No, the reader doesn't have such limitations.